### PR TITLE
feat: Language preferences in profile settings (behind feature flag)

### LINKED ISSUE #780: [Feature] Language preference selection in "Mitt AtB"
## Origin
_Links to received feedback, user research or other findings._

Default app criteria, English and Norwegian language support

## Motivation
_A short description of what user needs or business goals this feature will solve._

We only support English and Norwegian, and decide on which to use based on system language settings (phone settings). In cases where system language is set to something else than the supported languages, we choose a default on the users behalf. The user should be able to override this default language, so he/she doesn't need to change the language of their entire system to change preference for the app. 

## Hypotheses and assumptions
_A list of hypotheses and assumptions we have made about the user or the proposed solution._

- The user may be more familiar with the one or other of the supported languages, without using it as a system language on his/her phone
- Easily accessed language switch in app may enable the possibility for foreigners to practice their norwegian skills
- Learning mobility related terminology may be easier having an easily accessed translation in app

## Proposed solution
_A coarse description of a proposed solution, that may include wireframes and graphic design._

A switch for language in "Mitt AtB" screen, that explicitly sets system language settings. Option to use system settings if desired.

### PR DESCRIPTION
**Pull request by rikkekantega**
Defaults to "default language", currently set to "Norsk bokmål".

Using phone settings is optional - App then gets system locale, and decides on what will be considered the best app language.

User can choose app language explicitly from "Mitt AtB".

image

closes #780

### CODE REVIEW

File: src/translations/LanguageContext.tsx [hunk 1]
```diff
@@ -19,35 +20,62 @@ const AppLanguageProvider: React.FC = ({children}) => {
...
+    } else {
+      const onChange = () => {
+        setLocale(preferredLocale);
+      };
+      RNLocalize.addEventListener('change', onChange);
+      return () => {
+        RNLocalize.removeEventListener('change', onChange);
+      };
+    }
   }, []);
```
Comment (gorandalum) [inline src/translations/LanguageContext.tsx:50]: Shouldn't `shouldUseUserPreferenced` be in the dependency list here? Probably a corner case, but from my understanding the localization change listener is now only added if `shouldUseUserPreferenced` is false at initial mounting, and not if the user changes their setting to using the system language.

Adding `shouldUseUserPreferenced` will also make the `useEffect` on line 61 redundant.

**Review by gorandalum — APPROVED** [review]
Looks very good! And it feels good to finally review a change that is not 50 files, like most of the PRs the last two weeks (including mine) =)

File: src/translations/LanguageContext.tsx (hunk 1 above)
Comment (rikkekantega) [reply src/translations/LanguageContext.tsx:50]: Of course, not sure what I was even thinking. Good catch!

Comment (mikaelbr) [reply src/translations/LanguageContext.tsx]: I think maybe it's time to include dependency list eslint rule. I've also had some issues remembering all deps 😅

**Comment by mikaelbr** [comment]
A question that's maybe not directly related to this. We now feature toggle language selection, but the language is also selected by phone settings. Will people with `en` as default now see some English translations?

**Comment by rikkekantega** [comment]
> A question that's maybe not directly related to this. We now feature toggle language selection, but the language is also selected by phone settings. Will people with `en` as default now see some English translations?

The AppLanguageProvider considers the same feature toggle. :)

**Pull request merged**